userspace/engine/rule_loader_compiler.cpp

```diff
@@ -386,7 +386,11 @@ void rule_loader::compiler::compile_rule_infos(
 	std::string err, condition;
 	std::set<falco::load_result::load_result::warning_code> warn_codes;
 	filter_warning_resolver warn_resolver;
-	for (auto &r : col.rules())
+
+	// note: cast away the const qualifier in the for loop
+	// this is needed because we want to store information about evttypes
+	// used by any rules, which might come in handy when describing rules.
+	for (auto &r : const_cast<indexed_vector<rule_info>&>(col.rules()))
```

Comment:
Added information about evttypes as well. However, there is a little bit of a hack.
The collector returns rule infos in a const indexed vector. The only part of code that has information about evt codes (or the complete ast with all macros resolved) is this one. So we may want to add this information to rule_info, but we can't given the const qualifier. For this reason we a couple of options:
- stop using const. This is not considerable as a good option, since const correctness is correctly throughout this part of code, and I don't like remove it just for this use case. Removing it here would touch a chain of const object, so it would have effect also in other parts of the code
- use `const_cast` to cast away the const qualifier only in this case, and to not break the clean interface we have with these classes. This is the cleanest solution I found to this problem, happy to discuss alternatives if any!